### ISSUE TITLE
Crash during sync when hot restarting

### ISSUE THREAD

**Issue by MSKoWa** [aws-amplify/amplify-flutter#3574]
### Description

Hello,

I am receiving a crash when I hot restart my application on my Android device while debugging.

As you can see in the error below in appears among the different states of the initial sync. Relevant error might be:

`java.lang.IllegalStateException: Only a single observer allowed.`

 This ist the whole error before exiting:

```
[log] subscriptionsEstablished
I/amplify:aws-datastore(12701): Started subscription processor for models: [DataStoreTodo, DataStoreDosing, DataStoreMeasurement, DataStoreReefLight, DataStoreTankProfile, DataStoreTodoPlan, DataStoreParameter, DataStoreEvent] of types [ON_CREATE, ON_UPDATE, ON_DELETE].
[log] syncQueriesStarted
[log] toJson: AmplifyConfigState(user: Some(User(userId: Value(Right(670a1308-434e-4863-b535-9e0e59a65454)), userEmail: Value(Right(micsto231@gmail.com)))), syncFinished: false, initialized: true, agbAccepted: true, fetchUserFailureOption: None)
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
[log] modelSynced
I/amplify:aws-datastore(12701): Successfully sync'd down model state from cloud.
I/amplify:aws-datastore(12701): Starting processing subscription data buffer.
I/amplify:aws-datastore(12701): Started processing the mutation outbox. Pending mutations will be published to the cloud.
[log] modelSynced
W/amplify:aws-datastore(12701): Reading subscriptions buffer has failed.
W/amplify:aws-datastore(12701): java.lang.IllegalStateException: Only a single observer allowed.
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.subjects.UnicastSubject.subscribeActual(UnicastSubject.java:294)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnLifecycle.subscribeActual(ObservableDoOnLifecycle.java:33)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
[log] syncQueriesReady
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable.subscribeActual(ObservableFlatMapCompletableCompletable.java:49)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2836)
W/amplify:aws-datastore(12701): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.startDrainingMutationBuffer(SubscriptionProcessor.java:259)
W/amplify:aws-datastore(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$4$com-amplifyframework-datastore-syncengine-Orchestrator(Orchestrator.java:352)
W/amplify:aws-datastore(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator$$ExternalSyntheticLambda8.subscribe(Unknown Source:2)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/amplify:aws-datastore(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/amplify:aws-datastore(12701): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/amplify:aws-datastore(12701): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/amplify:aws-datastore(12701): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/amplify:aws-datastore(12701): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/amplify:aws-datastore(12701): 	at java.lang.Thread.run(Thread.java:919)
W/System.err(12701): io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalStateException: Only a single observer allowed.
W/System.err(12701): 	at io.reactivex.rxjava3.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
2
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:156)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable$FlatMapCompletableMainObserver.onError(ObservableFlatMapCompletableCompletable.java:125)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.observers.DisposableLambdaObserver.onError(DisposableLambdaObserver.java:65)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.disposables.EmptyDisposable.error(EmptyDisposable.java:63)
W/System.err(12701): 	at io.reactivex.rxjava3.subjects.UnicastSubject.subscribeActual(UnicastSubject.java:294)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnLifecycle.subscribeActual(ObservableDoOnLifecycle.java:33)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable.subscribeActual(ObservableFlatMapCompletableCompletable.java:49)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2836)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.startDrainingMutationBuffer(SubscriptionProcessor.java:259)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$4$com-amplifyframework-datastore-syncengine-Orchestrator(Orchestrator.java:352)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator$$ExternalSyntheticLambda8.subscribe(Unknown Source:2)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
[log] outboxStatus
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/System.err(12701): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/System.err(12701): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/System.err(12701): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/System.err(12701): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/System.err(12701): 	at java.lang.Thread.run(Thread.java:919)
W/System.err(12701): Caused by: java.lang.IllegalStateException: Only a single observer allowed.
W/System.err(12701): 	... 31 more
W/System.err(12701): io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalStateException: Only a single observer allowed.
W/System.err(12701): 	at io.reactivex.rxjava3.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
2
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:156)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable$FlatMapCompletableMainObserver.onError(ObservableFlatMapCompletableCompletable.java:125)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.observers.DisposableLambdaObserver.onError(DisposableLambdaObserver.java:65)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.disposables.EmptyDisposable.error(EmptyDisposable.java:63)
W/System.err(12701): 	at io.reactivex.rxjava3.subjects.UnicastSubject.subscribeActual(UnicastSubject.java:294)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnLifecycle.subscribeActual(ObservableDoOnLifecycle.java:33)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13099)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable.subscribeActual(ObservableFlatMapCompletableCompletable.java:49)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2836)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.startDrainingMutationBuffer(SubscriptionProcessor.java:259)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$startApiSync$4$com-amplifyframework-datastore-syncengine-Orchestrator(Orchestrator.java:352)
W/System.err(12701): 	at com.amplifyframework.datastore.syncengine.Orchestrator$$ExternalSyntheticLambda8.subscribe(Unknown Source:2)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2850)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
W/System.err(12701): 	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/System.err(12701): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/System.err(12701): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/System.err(12701): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/System.err(12701): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/System.err(12701): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/System.err(12701): 	at java.lang.Thread.run(Thread.java:919)
W/System.err(12701): Caused by: java.lang.IllegalStateException: Only a single observer allowed.
W/System.err(12701): 	... 31 more
E/memtrack(12701): Couldn't load memtrack module
W/android.os.Debug(12701): failed to get memory consumption info: -1
I/Process (12701): Sending signal. PID: 12701 SIG: 9
Lost connection to device.
Exited
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.6

### Amplify Flutter Version

1.3.2

### Deployment Method

Amplify CLI

**Comment by fjnoyp** [comment]
HI @MSKoWa thanks for sharing your issue.  Can you please share your code that initializes Amplify and triggers this error?  Without these details it's hard for us to reproduce your issue on our side.  Based on the error message it seems you might be adding multiple observers somewhere, so providing any code related to that will help this investigation as well.

**Comment by MSKoWa** [comment]
Hello,

this is part of my schema:

```
type DataStoreEvent @model @auth(rules: [{ allow: private }]) {
  creator: String @index(name: "byCreator", sortKeyFields: ["id"])
  owner: String! @index(name: "byOwner", sortKeyFields: ["id"])
  id: ID!
  eventType: String!
  dateTime: AWSDateTime!
  note: String!
  datastoretankprofileID: ID! @index(name: "byDataStoreTankProfile")
}

type DataStoreMeasurement @model @auth(rules: [{ allow: private }]) {
  creator: String @index(name: "byCreator", sortKeyFields: ["id"])
  owner: String! @index(name: "byOwner", sortKeyFields: ["id"])
  source: String @index(name: "bySource", sortKeyFields: ["id"])
  id: ID!
  value: Float!
  dateTime: AWSDateTime!
  temperatureValue: Float
  datastoreparameterID: ID! @index(name: "byDataStoreParameter")
}

type DataStoreParameter @model @auth(rules: [{ allow: private }]) {
  creator: String @index(name: "byCreator", sortKeyFields: ["id"])
  owner: String! @index(name: "byOwner", sortKeyFields: ["id"])
  id: ID!
  parameterType: String!
  upperBound: Float!
  lowerBound: Float!
  onDashboard: Boolean!
  datastoretankprofileID: ID! @index(name: "byDataStoreTankProfile")
  DataStoreMeasurements: [DataStoreMeasurement]
    @hasMany(indexName: "byDataStoreParameter", fields: ["id"])
}

type DataStoreTankProfile @model @auth(rules: [{ allow: private }]) {
  owner: String! @index(name: "byOwner", sortKeyFields: ["id"])
  access: [String]
  id: ID!
  tankType: String!
  name: String!
  volume: Float
  imageName: String
  creationDate: AWSDateTime
  DataStoreParameters: [DataStoreParameter]
    @hasMany(indexName: "byDataStoreTankProfile", fields: ["id"])
  DataStoreEvents: [DataStoreEvent]
    @hasMany(indexName: "byDataStoreTankProfile", fields: ["id"])
}
```

This is the essential code of my Amplify initialization where the state.user is an Option<String>:

```
final AmplifyDataStore _dataStorePlugin = AmplifyDataStore(
      modelProvider: ModelProvider.instance,
      syncExpressions: [
        DataStoreSyncExpression(
            DataStoreTankProfile.classType,
            () => DataStoreTankProfile.OWNER.eq(
                state.user.fold(() => "", (user) => user))),
        DataStoreSyncExpression(
            DataStoreParameter.classType,
            () => DataStoreParameter.OWNER.eq(
                state.user.fold(() => "", (user) => user))),
        DataStoreSyncExpression(
            DataStoreMeasurement.classType,
            () => DataStoreMeasurement.OWNER.eq(
                state.user.fold(() => "", (user) => user))),
        DataStoreSyncExpression(
            DataStoreEvent.classType,
            () => DataStoreEvent.OWNER.eq(
                state.user.fold(() => "", (user) => user))),
      ],
    );

    final AmplifyAPI _apiPlugin = AmplifyAPI();
    final AmplifyAuthCognito _authPlugin = AmplifyAuthCognito();
    final AmplifyStorageS3 _storagePlugin = AmplifyStorageS3();

    try {
      if (!amplify.isConfigured) {
        await amplify.addPlugins([
          _dataStorePlugin,
          _apiPlugin,
          _authPlugin,
          _storagePlugin,
        ]);
        await amplify.configure(amplifyconfig);
      }
    } on AmplifyAlreadyConfiguredException catch(e){
      log("Amplify still wanted to configure itself event it is configured....");
    }
```

I do indeed have multiple observers because I have mutliple models to observe and to interact with added like this:

```
@override
  Stream<List<TankProfileDto>> watchAllTankProfiles(String owner) async* {
    final where = DataStoreTankProfile.OWNER.eq(owner);
    final query = () =>
        amplify.DataStore.query(DataStoreTankProfile.classType, where: where);

    yield (await query())
        .map((tankProfile) => TankProfileDto.fromDataStore(tankProfile, none()))
        .toList();

    yield* amplify.DataStore.observe(DataStoreTankProfile.classType,
            where: where)
        .asyncMap((event) async => (await query()).map((tankProfile) {
              return TankProfileDto.fromDataStore(tankProfile, none());
            }).toList());
  }
```

 Is this a problem?

On IOS I do not have any of these issues.

Best regards

**Comment by fjnoyp** [comment]
Hi @MSKoWa thanks for sharing more details.  I tried reproducing your crash by calling hot restart in our basic Datastore example app with `DataStoreSyncExpression` and calling `observe` but no error or crash occurred. 

Calling query or observe or having sync expressions shouldn't be a problem with hot restart.  

Let's try to narrow down the possible causes for your crash.  Within your IDE can you please debug your app and then trigger the crash by doing a hot restart.  In debug mode, your app should automatically breakpoint at the line of code that failed.  Normally this is from the framework, but you should see a stack trace option that allows you to see the code that triggered the crash.  More context [here](https://docs.flutter.dev/testing/debugging).  Could you do that and then provide the stack trace - this way we can see what line of code in your file called which Amplify Datastore method that triggered the exception.

**Comment by MSKoWa** [comment]
Ah yeah, here we go:

```
DataStoreException (DataStoreException {
  "message": "Failed to start DataStore.",
  "recoverySuggestion": "Retry.",
  "underlyingException": "DataStoreException{message=Timed out while starting to observe storage changes., cause=com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Not a JSON Array: null, recoverySuggestion=There is a possibility that there is a bug if this error persists. Please take a look at \nhttps://github.com/aws-amplify/amplify-android/issues to see if there are any existing issues that \nmatch your scenario, and file an issue with the details of the bug if there isn't.}"
})

```

which occurs when we call `await amplify.DataStore.start();`


I appreciate your help!

**Comment by ridakaddir** [comment]
this exception was mentioned in this thread 

https://github.com/aws-amplify/amplify-flutter/issues/3017#issuecomment-1636064284

**Comment by Jordan-Nelson** [comment]
I am going to close this out as a duplicate of #3017. This should be resolved in v1.4.0
